src/test_utils.rs

```diff
@@ -2,17 +2,21 @@
 // so that tests can be run with all the necessary resources.
 // This removes the requirement for having to limit test threads
 // or to build/run nym-client ourselves.
-#[macro_export]
-macro_rules! new_nym_client {
-    ($nym_id:ident, $uri:ident) => {
-        let docker_client = clients::Cli::default();
-        let nym_ready_message = WaitFor::message_on_stderr("Client startup finished!");
-        let nym_image = GenericImage::new("nym", "latest")
-            .with_env_var("NYM_ID", $nym_id)
-            .with_wait_for(nym_ready_message)
-            .with_exposed_port(1977);
-        let nym_container = docker_client.run(nym_image);
-        let nym_port = nym_container.get_host_port_ipv4(1977);
-        let $uri = format!("ws://0.0.0.0:{nym_port}");
-    };
+
+use testcontainers::{clients::Cli, core::WaitFor, images::generic::GenericImage, Container};
+
+/// Create a nym client using the same docker Cli
+pub fn create_nym_client<'a>(
+    docker_client: &'a Cli,
+    nym_id: &str,
+) -> (Container<'a, GenericImage>, u16, String) {
+    let nym_ready_message = WaitFor::message_on_stderr("Client startup finished!");
+    let nym_image = GenericImage::new("nym", "latest")
+        .with_env_var("NYM_ID", nym_id)
+        .with_wait_for(nym_ready_message)
+        .with_exposed_port(1977);
+    let nym_container = docker_client.run(nym_image);
+    let nym_port = nym_container.get_host_port_ipv4(1977);
+    let nym_uri = format!("ws://0.0.0.0:{nym_port}");
+    (nym_container, nym_port, nym_uri)
```

Comment:
Not at this time, but I was thinking of spawning up a docker image with a deliberately configured to have a slow network connection to test the nym messages arriving out of order. This may not be necessary atm, but I will experimenting with it, and starting with libp2p first where ports are needed to map with which containers.

Reply:
I think we can remove the `nym_port` for now since it's unused everywhere, and add it in if necessary later. better to not have unused vars imo

Reply:
also, the container also appears to be unused everywhere (unless I missed something), is it necessary to return it? I assume for lifetime purposes?

Reply:
@noot, yes the container has to be returned so it doesn't go out of scope prematurely.